x-pack/examples/third_party_maps_source_example/public/classes/custom_raster_source.tsx

```diff
@@ -42,7 +42,24 @@ export class CustomRasterSource implements IRasterSource {
   constructor(sourceDescriptor: CustomRasterSourceDescriptor) {
     this._descriptor = sourceDescriptor;
   }
+  async hasLegendDetails(): Promise<boolean> {
+    return true;
+  }
 
+  renderLegendDetails(): ReactElement<any> | null {
+    return (
+      <div
+        style={{
+          height: '1em',
+          background:
+            'linear-gradient(267deg, rgb(34, 84, 61) 0%, rgb(55, 124, 89) 12.67%, rgb(104, 211, 145) 38%, rgb(109, 219, 195) 46.67%, rgb(190, 227, 248) 64%, rgb(123, 197, 244) 76%, rgb(49, 130, 206) 100%)',
+        }}
+      >
+        <span style={{ position: 'absolute', left: '0' }}>less</span>{' '}
+        <span style={{ position: 'absolute', right: '0' }}>more</span>
+      </div>
+    );
```

Comment:
I've discovered that [NOAA also publishes a legend for their services](https://nowcoast.noaa.gov/help/#!section=layerinfo) that we could use. What do you think?

```suggestion
    return <img alt="Radar legend" src="https://nowcoast.noaa.gov/images/legends/radar.png" />;
```

Reply:
More than happy to use the image. Its cleaner than what I threw together in a few seconds.